kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -0,0 +1,456 @@
+        memory_capacity = node_spec.get('memory_capacity', 0)
+
+        tags = instance_tags
+        self.gauge(self, self.NAMESPACE + '.cpu.capacity', float(num_cores), tags)
+        self.gauge(self, self.NAMESPACE + '.memory.capacity', float(memory_capacity), tags)
+
+    def _perform_kubelet_check(self, instance_tags):
+        """Runs local service checks"""
+        service_check_base = self.NAMESPACE + '.kubelet.check'
+        is_ok = True
+        url = self.kube_health_url
+
+        try:
+            req = self.perform_kubelet_query(url)
+            for line in req.iter_lines():
+                # avoid noise; this check is expected to fail since we override the container hostname
+                if line.find('hostname') != -1:
+                    continue
+
+                matches = re.match(r'\[(.)\]([^\s]+) (.*)?', line)
+                if not matches or len(matches.groups()) < 2:
+                    continue
+
+                service_check_name = service_check_base + '.' + matches.group(2)
+                status = matches.group(1)
+                if status == '+':
+                    self.service_check(service_check_name, AgentCheck.OK, tags=instance_tags)
+                else:
+                    self.service_check(service_check_name, AgentCheck.CRITICAL, tags=instance_tags)
+                    is_ok = False
+
+        except Exception as e:
+            self.log.warning('kubelet check %s failed: %s' % (url, str(e)))
+            self.service_check(service_check_base, AgentCheck.CRITICAL,
+                               message='Kubelet check %s failed: %s' % (url, str(e)), tags=instance_tags)
+        else:
+            if is_ok:
+                self.service_check(service_check_base, AgentCheck.OK, tags=instance_tags)
+            else:
+                self.service_check(service_check_base, AgentCheck.CRITICAL, tags=instance_tags)
+
+    def _report_pods_running(self, pods, instance_tags):
+        """
+        Reports the number of running pods on this node
+        tagged by service and creator.
+        """
+        for pod in pods['items']:
+            pod_id = pod.get('metadata', {}).get('uid')
+            tags = get_tags('kubernetes_pod://%s' % pod_id) or None
+            if not tags:
+                continue
+            self.gauge(self, self.NAMESPACE + '.pods.running', 1, tags)
+
+    def _report_container_spec_metrics(self, pod_list, instance_tags):
+        """Reports pod requests & limits by looking at pod specs."""
+        for pod in pod_list['items']:
+            pod_meta = pod.get('metadata', {})
+            _, pod_name = pod_meta.get('namespace'), pod_meta.get('name')
+
+            if not pod_name:
+                continue
+
+            for ctr in pod['spec']['containers']:
+                if not ctr.get('resources'):
+                    continue
+
+                c_name = ctr.get('name', '')
+                tags = get_tags('docker://%s' % ctr['id'])
```

Comment:
The safest way to the the container's entity ID is to use `items.status.containerStatuses[].containerID`. It's where the kubeletcollector takes them. Format is already `"docker://710695aa82cb808e979e39078f6dd18ece04d2bf444fdf78e9b37e360b6882d5"`

You can then match the containers from `spec` via the `name` field, that has unique value inside a given pod.

You'll want to add the second argument for cardinality, probably `true`